resources/js/views/rooms/Index.vue

```diff
@@ -50,49 +81,73 @@ export default {
       currentUser: state => state.session.currentUser
     }),
     showLimit: function () {
-      return this.currentUser.room_limit !== -1 && this.rooms.myRooms !== undefined;
+      return this.currentUser.room_limit !== -1 && this.ownRooms !== null;
     },
     limitReached: function () {
-      return this.currentUser.room_limit !== -1 && this.rooms.myRooms !== undefined && this.rooms.myRooms.length >= this.currentUser.room_limit;
+      return this.currentUser.room_limit !== -1 && this.ownRooms !== null && this.ownRooms.data.length >= this.currentUser.room_limit;
+    },
+    searchQuery: function () {
+      return this.rawSearchQuery.trim() === '' ? '' : '&search=' + this.rawSearchQuery.trim();
     }
   },
+  mounted: function () {
+    this.reload();
+  },
   methods: {
     // Handle event from new room component that the limit was reached
     onReachLimit () {
       store.dispatch('session/getCurrentUser');
-      this.reload();
+      this.loadOwnRooms();
     },
+    // Load all required resources
     reload () {
-      Base.call('rooms').then(response => {
-        this.rooms = response.data.data;
-      }).catch((error) => {
-        Base.error(error, this.$root);
+      this.loadOwnRooms();
+      this.loadSharedRooms();
+      this.loadRoomTypes();
+    },
+    // Reset page of pagination and reload resources with search query
+    search () {
+      this.ownRooms.meta.current_page = 1;
+      this.sharedRooms.meta.current_page = 1;
+      this.loadOwnRooms();
+      this.loadSharedRooms();
+    },
+    // Load the rooms shared with the current user
+    loadSharedRooms () {
+      this.loadingShared = true;
+      const page = this.ownRooms !== null ? this.ownRooms.meta.current_page : 1;
+      Base.call('rooms?filter=shared&page=' + page + this.searchQuery).then(response => {
```

Comment:
Query parameters can be passed through the config object, which is the second parameter of this function!

Reply:
oh, great